Userland/Utilities/pls.cpp

```diff
@@ -0,0 +1,196 @@
+    auto paths = String(getenv("PATH")).split(':');
+    int num_unveils = 0;
+    char path_buf[256];
+
+    // Unveil each path
+    for (const auto path : paths) {
+        if (unveil(path.characters(), "x") == 0)
+            num_unveils++;
+    }
+
+    // Now unveil the command
+    auto command_path = realpath(command, &path_buf[0]);
+    if (command_path) {
+        if (unveil(command_path, "x") == 0)
+            num_unveils++;
+    }
+
+    return num_unveils;
+}
+
+// <kling> linusg: quaker: "please" feels quite long, how about "pls" :P
+// <kling> "pls rm -r crap" has a nice ring to it
+// lol
+int main(int argc, char** argv)
+{
+    Vector<const char*> command;
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(command, "Name of command to run at elevated privilege level", "cmd");
+
+    args_parser.parse(argc, argv);
+
+    if (pledge("stdio tty rpath exec id", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    if (seteuid(0) < 0) {
+        perror("seteuid");
+        return 1;
+    }
+
+    if (unveil("/etc", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    const char* username = getlogin();
+    Vector<String> sudoers;
+    auto sudoers_fd = fopen("/etc/sudoers", "r");
+    if (!sudoers_fd) {
+        warnln("couldn't open /etc/sudoers!");
+        return 1;
+    }
+
+    // Okay, so we have a handle to /etc/sudoers, now let's read it in line by line to get a
+    // list of all of the users contained within
+    char buffer[line_buff_length];
+    while (fgets(buffer, line_buff_length, sudoers_fd)) {
+        // Ignore comments
+        if (buffer[0] == '#')
+            continue;
+
+        buffer[strcspn(buffer, "\n")] = '\0';
+        sudoers.append(String(buffer));
+    }
+
+    if (!sudoers.contains_slow(username)) {
+        warnln("{} is not in the sudoers file. This incident will be reported :^)", username);
+        return 2;
+    }
+
+    // The user was in the sudoers file, now let's ask for their password to ensure that it's actually them...
+    uid_t current_uid = getuid();
+    auto account_or_error = (username)
+        ? Core::Account::from_name(username)
+        : Core::Account::from_uid(current_uid);
+
+    if (account_or_error.is_error()) {
+        fprintf(stderr, "Core::Account::from_name: %s\n", account_or_error.error().characters());
+        return 1;
+    }
+
+    const auto& account = account_or_error.value();
+    if (getuid() != 0 && account.has_password()) {
+        auto password = Core::get_password();
+        if (password.is_error()) {
+            warnln("{}", password.error());
+            return 1;
+        }
+
+        if (!account.authenticate(password.value().characters())) {
+            warnln("Incorrect or disabled password.");
+            return 1;
+        }
+    }
+
+    // TODO: Support swapping users instead of just defaulting to root
+    setuid(0);
+    setgid(0);
```

Comment:
This PR seems like quite a party 🎉. I believe you're supposed to set group before user, normally 😁.